index.bs

```diff
@@ -212,6 +212,32 @@ examples that violate the above rules are <code>XMLHttpRequest</code> and
 initialisms, even if they are repeated.
 </div>
 
+<h3 id="prefer-dict-to-bool">Prefer dictionary parameters over boolean parameters or other unreadable parameters</h3>
+
+APIs should generally prefer dictionary parameters over booleans.
+This makes the code that calls the API <a href="https://ariya.io/2011/08/hall-of-api-shame-boolean-trap">much more readable</a>.
+It also makes the API more extensible in the future,
+particularly if multiple booleans are needed.
+
+<p class="example">For example,
+<code highlight="js">new Event("exampleevent", { bubbles: true, cancelable: false})</code>
+is much more readable than
+<code highlight="js">new Event("exampleevent", true, false)</code>.
+
+Furthermore,
+the booleans in dictionaries need to be designed so that they all default to false.
+If booleans default to true, then
+<a href="https://lists.w3.org/Archives/Public/public-script-coord/2013OctDec/0302.html">users of the API will find unexpected JavaScript behavior</a> since <code highlight="js">{ passive: false }</code> and <code highlight="js">{ passive: undefined }</code> will produce different results.
+But at the same time, it's important to avoid naming booleans in negative ways,
+because then code will have confusing double-negatives.
```

Comment:
Added: These pieces of advice may sometimes conflict, but the conflict can be avoided by using opposite words without negatives, such as “repeat” versus “once”, “isolate” versus “connect”, or “private” versus “public”.